pkg/placement/what-resolver.go

```diff
@@ -639,22 +639,32 @@ func whatMatchingPlacements(logger klog.Logger, wsd *workspaceDetails, candidate
 
 // returns `(changed bool, success bool)`
 func (od *objectDetails) setByMatch(logger klog.Logger, wsd *workspaceDetails, spec *edgeapi.EdgePlacementSpec, epName ObjectName, whatResource string, whatObj mrObject) (bool, bool) {
-	wantSingletonReturn, found := od.PlacementBits.Get(epName)
+	oldDistrBits, found := od.PlacementBits.Get(epName)
+	newDistrBits := DistributionBits{ReturnSingletonState: spec.WantSingletonReportedState,
+		CreateOnly: whatObj != nil && isCreateOnly(whatObj)}
```

Comment:
Not crucial - if the next condition doesn't depend on the above oldDistrBits and newDistrBits, why not moving this to after the if ! success ?

Reply:
Yes, there is some freedom in the ordering of these statements.